packages/common/src/api/tan-query/comments.ts

```diff
@@ -0,0 +1,719 @@
+import { useAudiusQueryContext } from '~/audius-query'
+import { Comment, ID, ReplyComment } from '~/models'
+import { toast } from '~/store/ui/toast/slice'
+import { encodeHashId } from '~/utils'
+
+type CommentOrReply = Comment | ReplyComment
+
+const QUERY_KEYS = {
+  trackCommentList: 'trackCommentList',
+  comment: 'comment',
+  replies: 'replies'
+} as const
+
+const messages = {
+  loadError: (type: 'comments' | 'replies') =>
+    `There was an error loading ${type}. Please try again.`,
+  mutationError: (
+    actionType:
+      | 'pinning'
+      | 'unpinning'
+      | 'deleting'
+      | 'posting'
+      | 'editing'
+      | 'reacting to'
+      | 'reporting'
+  ) => `There was an error ${actionType} that comment. Please try again`,
+  muteUserError: 'There was an error muting that user. Please try again.'
+}
+
+/**
+ *
+ * QUERIES
+ *
+ */
+
+type GetCommentsByTrackArgs = {
+  trackId: ID
+  userId: ID | null
+  sortMethod: CommentSortMethod
+  pageSize?: number
+}
+
+export const useGetCommentsByTrackId = ({
+  trackId,
+  userId,
+  sortMethod,
+  pageSize = 5
+}: GetCommentsByTrackArgs) => {
+  const { audiusSdk, reportToSentry } = useAudiusQueryContext()
+  const isMutating = useIsMutating()
+  const queryClient = useQueryClient()
+  const queryRes = useInfiniteQuery({
+    enabled: !!userId && !!trackId && isMutating === 0,
+    getNextPageParam: (lastPage: ID[], pages) => {
+      if (lastPage?.length < pageSize) return undefined
+      return (pages.length ?? 0) * pageSize
+    },
+    queryKey: [QUERY_KEYS.trackCommentList, trackId, sortMethod],
+    queryFn: async ({ pageParam: currentPage = 0 }): Promise<ID[]> => {
+      const sdk = await audiusSdk()
+      const commentsRes = await sdk.tracks.trackComments({
+        trackId: encodeHashId(trackId),
+        offset: currentPage,
+        limit: pageSize,
+        sortMethod,
+        userId: userId?.toString() ?? undefined
+      })
+      const commentList = transformAndCleanList(
+        commentsRes.data,
+        commentFromSDK
+      )
+
+      // Populate individual comment cache
+      commentList.forEach((comment) => {
+        queryClient.setQueryData([QUERY_KEYS.comment, comment.id], comment)
+        comment?.replies?.forEach?.((reply: ReplyComment) =>
+          queryClient.setQueryData([QUERY_KEYS.comment, reply.id], reply)
+        )
+      })
+      // For the comment list cache, we only store the ids of the comments (organized by sort method)
+      return commentList.map((comment) => comment.id)
+    },
+    onError: (error: Error) => {
+      reportToSentry({
+        error,
+        name: 'Comments'
+      })
+      toast({ content: messages.loadError('replies') })
+    },
+    staleTime: Infinity, // Stale time is set to infinity so that we never reload data thats currently shown on screen (because sorting could have changed)
+    cacheTime: 1 // Cache time is set to 1 so that the data is cleared any time we leave the page viewing it or change sorts
```

Comment:
TODO for myself - in hindsight since we're not caching sort data I dont need to store by `currentSort` at all; I just need the currently active data. Simplifies having to pass `currentSort` into every method lol

Reply:
Oh true, duh, thats why I put the sort key in the first place 🙈 
yeah no it needs to stay lol